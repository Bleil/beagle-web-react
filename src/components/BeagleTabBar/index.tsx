import React, { FC } from 'react'
import { BeagleComponent } from '@zup-it/beagle-react'
import {
  StyledTabBar,
  StyledBeagleTabItem,
  StyledBeagleTabItemContent,
  StyledSelected,
  StyledUnselect,
} from './styled'

export interface ItemTitle {
  title?: string,
  icon?: ImagePath,
}

export interface ImagePath {
  _beagleImagePath_: 'local' | 'remote',
  url: string,
}

export interface BeagleTabBarInterface extends BeagleComponent {
  onTabSelection?: (item: number) => void,
  currentTab?: number,
  items: ItemTitle[],
  styleId?: string,
}

const BeagleTabBar: FC<BeagleTabBarInterface> = ({
  onTabSelection, currentTab, items, beagleContext, styleId,
}) => {

  const changeSelectedTab = (index: number) => {
    if (!onTabSelection) return
    onTabSelection(index)
  }

  return (
    <StyledTabBar className={styleId}>
      {items.map((item, index) => (
        <StyledBeagleTabItem key={index}>

          <StyledBeagleTabItemContent onClick={() => changeSelectedTab(index)}>
            {index === currentTab && <StyledSelected>{item.title}</StyledSelected>}
            {index !== currentTab && <StyledUnselect>{item.title}</StyledUnselect>}
          </StyledBeagleTabItemContent >
        </StyledBeagleTabItem>
      ))}
    </StyledTabBar>
  )
}

export default BeagleTabBar