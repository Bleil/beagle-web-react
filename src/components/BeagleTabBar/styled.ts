import styled from 'styled-components'

export const StyledTabBar = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width:100%;
  background-color:#c5c5c5;
  border-radius:50px;
  color: #ffffff;
`

export const StyledBeagleTabItem = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width:50%;
`

export const StyledBeagleTabItemContent = styled.div`
  width:100%
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  min-width: 20px;
  min-height: 15px;
`


export const StyledBeagleTabItemSelected = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width:50%;
  background-color: #df8006;
`


export const StyledSelected = styled.div`
  background-color: #df8006;
`


export const StyledUnselect = styled.div`
  background-color: #c5c5c5;
`
