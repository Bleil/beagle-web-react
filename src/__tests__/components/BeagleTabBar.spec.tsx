/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

// Link.react.test.js
import * as React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure, mount } from 'enzyme'
import mock from 'jest-mock-extended/lib/Mock'
import BeagleTabBar, { ItemTitle } from '../../components/BeagleTabBar'
import { BeagleContext } from '../../types'

let beagleContextMock: any
let wrapper: any
let onTabSelectionMock: any
const itemTitleMock: ItemTitle[] = [{ title:'Title A' }, { title:'Title B' }, { title:'Title C' }]

configure({ adapter: new Adapter() })
beforeAll(() => {
  beagleContextMock = mock<BeagleContext>()
  onTabSelectionMock = jest.fn(()=>'submited')
  wrapper = mount(<BeagleTabBar 
    items={itemTitleMock} 
    beagleContext={beagleContextMock} 
    onTabSelection={onTabSelectionMock}/>)
})

test('Beagle snapshot TabBar', () => {
  expect(wrapper).toMatchSnapshot()
})
