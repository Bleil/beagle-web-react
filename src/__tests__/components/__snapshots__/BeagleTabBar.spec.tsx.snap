// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beagle snapshot TabBar 1`] = `
<ComponentWithTheme
  items={
    Array [
      Object {
        "title": "Title A",
      },
      Object {
        "title": "Title B",
      },
      Object {
        "title": "Title C",
      },
    ]
  }
  onTabSelection={[MockFunction]}
>
  <BeagleTabBar
    className=""
    items={
      Array [
        Object {
          "title": "Title A",
        },
        Object {
          "title": "Title B",
        },
        Object {
          "title": "Title C",
        },
      ]
    }
    onTabSelection={[MockFunction]}
  >
    <styled.div>
      <div
        className="sc-AxiKw dPWazA"
      >
        <styled.div
          key="0"
        >
          <div
            className="sc-AxhCb kOGcQs"
          >
            <styled.div
              onClick={[Function]}
            >
              <div
                className="sc-AxhUy kHJxlU"
                onClick={[Function]}
              >
                Title A
              </div>
            </styled.div>
          </div>
        </styled.div>
        <styled.div
          key="1"
        >
          <div
            className="sc-AxhCb kOGcQs"
          >
            <styled.div
              onClick={[Function]}
            >
              <div
                className="sc-AxhUy kHJxlU"
                onClick={[Function]}
              >
                Title B
              </div>
            </styled.div>
          </div>
        </styled.div>
        <styled.div
          key="2"
        >
          <div
            className="sc-AxhCb kOGcQs"
          >
            <styled.div
              onClick={[Function]}
            >
              <div
                className="sc-AxhUy kHJxlU"
                onClick={[Function]}
              >
                Title C
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </BeagleTabBar>
</ComponentWithTheme>
`;
